src/Halogen/VDom/Types.purs

```diff
@@ -39,6 +40,19 @@ instance bifunctorVDom ∷ Bifunctor VDom where
   bimap f g (Grafted a) = Grafted (bimap f g a)
   bimap f g a = Grafted (graft (Graft f g a))
 
+-- | Replaces "widgets" in the `VDom` with the ability to turn them into other
+-- | `VDom` nodes.
+-- |
+-- | Using this function will fuse any `Graft`s present in the `VDom`.
+renderWidget ∷ ∀ a w x. (w → VDom a x) → VDom a w → VDom a x
+renderWidget f = case _ of
+  Text a → Text a
+  Elem ns n a ch → Elem ns n a (map (renderWidget f) ch)
+  Keyed ns n a ch → Keyed ns n a (map (map (renderWidget f)) ch)
+  Widget w → f w
+  Grafted g → Grafted $ unGraft (\(Graft fa fw v) →
+    graft (Graft identity identity (renderWidget f (bimap fa fw v)))) g
```

Comment:
Actually, this would probably be better as `Grafted g → renderWidget f (runGraft g)` I guess?